src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1898,21 +1898,16 @@ bool QgsOgrProvider::addFeatures( QgsFeatureList &flist, Flags flags )
   const bool inTransaction = startTransaction();
 
   QgsFeatureId incrementalFeatureId = -1;
-  OGRGeometryH filter;
   if ( !( flags & QgsFeatureSink::FastInsert ) &&
        ( mGDALDriverName == QLatin1String( "CSV" ) || mGDALDriverName == QLatin1String( "XLSX" ) || mGDALDriverName == QLatin1String( "ODS" ) ) )
   {
-    filter = mOgrLayer->GetSpatialFilter();
-    if ( filter )
+    QMutex *mutex = nullptr;
+    OGRLayerH layer = mOgrOrigLayer->getHandleAndMutex( mutex );
     {
-      filter = OGR_G_Clone( filter );
-      mOgrLayer->SetSpatialFilter( nullptr );
+      QMutexLocker locker( mutex );
+      OGR_L_SetAttributeFilter( layer, nullptr );
```

Comment:
shouldn't the filter, if set, be restored after the additions are finished ?

Reply:
Actually, I'm not even sure setting attribute filter to null is needed here. Is the original layer reference always filter-free?

Reply:
> Is the original layer reference always filter-free?

from my understanding of QgsOgrProvider::_setSubsetString(), no, if a subset string has been defined, either in the URI or through setSubsetString()

Reply:
Ok, I'll adjust accordingly.